static/app/views/starfish/views/webServiceView/starfishView.tsx

```diff
@@ -1,177 +1,174 @@
-
-import {getInterval} from 'sentry/components/charts/utils';
-import {t} from 'sentry/locale';
+import {EventsStats} from 'sentry/types';
+import {Series, SeriesDataUnit} from 'sentry/types/echarts';
 import {tooltipFormatterUsingAggregateOutputType} from 'sentry/utils/discover/charts';
+import EventView from 'sentry/utils/discover/eventView';
 import {RateUnits} from 'sentry/utils/discover/fields';
 import {DiscoverDatasets} from 'sentry/utils/discover/types';
 import {formatRate} from 'sentry/utils/formatters';
+import {usePageError} from 'sentry/utils/performance/contexts/pageError';
 import {MutableSearch} from 'sentry/utils/tokenizeSearch';
 import usePageFilters from 'sentry/utils/usePageFilters';
-import withApi from 'sentry/utils/withApi';
 import {AVG_COLOR, THROUGHPUT_COLOR} from 'sentry/views/starfish/colours';
 import Chart, {useSynchronizeCharts} from 'sentry/views/starfish/components/chart';
 import MiniChartPanel from 'sentry/views/starfish/components/miniChartPanel';
 import {STARFISH_CHART_INTERVAL_FIDELITY} from 'sentry/views/starfish/utils/constants';
-import {DataTitles} from 'sentry/views/starfish/views/spans/types';
+import {useEventsStatsQuery} from 'sentry/views/starfish/utils/useEventsStatsQuery';
+import {DataTitles, getThroughputTitle} from 'sentry/views/starfish/views/spans/types';
 import {SpanGroupBar} from 'sentry/views/starfish/views/webServiceView/spanGroupBar';
 import {BaseStarfishViewProps} from 'sentry/views/starfish/views/webServiceView/starfishLanding';
 
 import EndpointList from './endpointList';
 
 export function StarfishView(props: BaseStarfishViewProps) {
-  const {organization, eventView} = props;
   const pageFilter = usePageFilters();
-  const theme = useTheme();
-  const [isLoading, setIsLoading] = useState(true);
+  const {selection} = pageFilter;
+  const [activeSpanGroup, setActiveSpanGroup] = useState<string | null>(null);
 
-  function renderCharts() {
-    const query = new MutableSearch([
+  function useRenderCharts() {
+    let query = new MutableSearch([
       'event.type:transaction',
       'has:http.method',
       'transaction.op:http.server',
     ]);
+    let dataset = DiscoverDatasets.METRICS;
+    let yAxis = ['tps()', 'http_error_count()', 'avg(transaction.duration)'];
+    let titles = [DataTitles.throughput, DataTitles.errorCount, DataTitles.avg];
+    if (activeSpanGroup) {
+      query = new MutableSearch([
+        'transaction.op:http.server',
+        `span.module:${activeSpanGroup}`,
+      ]);
+      dataset = DiscoverDatasets.SPANS_METRICS;
+      yAxis = ['sps()', 'http_error_count()', 'avg(span.self_time)'];
+      titles = [
+        getThroughputTitle(activeSpanGroup),
+        activeSpanGroup === 'http' ? DataTitles.errorCount : '--',
+        `Avg Duration of ${activeSpanGroup} ${
+          activeSpanGroup === 'db' ? 'Queries' : 'Spans'
+        }`,
+      ];
+    }
+
+    const {isLoading: loading, data: results} = useEventsStatsQuery({
```

Comment:
had to rewrite this using `useEventsStatsQuery` so that we wouldn't requery on hover change